### PR TITLE
feat: add allowClearnetHttpRequests feature flag

### PR DESCRIPTION
**Pull request by wodoro**
Allows safe use of clear net price feed during development

You can add `--allowClearnetHttpRequests=true` to your launch params and price feed will work again in dev env

## Summary by CodeRabbit

* **New Features**
  * New runtime option to permit clearnet HTTP fallback when no SOCKS5/Tor proxy is available, with safer routing and explicit IP-leak warnings (esp. on mainnet).
  * Improved HTTP handling to refuse onion targets without a proxy.

* **Tests**
  * Added thorough tests for onion-host detection, including edge cases and format variants.

* **Documentation**
  * Updated developer setup guide with the new HTTP configuration options.

[Review Change Stack](https://app.coderabbit.ai/change-stack/bisq-network/bisq/pull/7727)

### CODE REVIEW

File: common/src/main/java/bisq/common/config/Config.java [hunk 1]
```diff
@@ -597,6 +599,17 @@ public Config(String defaultAppName, File defaultUserDataDir, String... args) {
...
+                parser.accepts(ALLOW_CLEARNET_HTTP_REQUESTS,
+                                "If true, HTTP requests to non-local destinations are sent directly " +
+                                        "over the clearnet, bypassing Tor. Intended for development and " +
+                                        "testing against a known endpoint (e.g. a self-hosted price node). " +
+                                        "WARNING: leaks your IP address to the destination server. " +
+                                        "Do not enable on mainnet with real funds.")
+                        .withRequiredArg()
+                        .ofType(Boolean.class)
+                        .defaultsTo(false);
+
```
[the remark this answers is not in the document]
Comment (wodoro) [reply common/src/main/java/bisq/common/config/Config.java:612]: User may have legitmate use cases, with good understanding of the risks.

File: core/src/main/java/bisq/core/app/BisqSetup.java [hunk 2]
```diff
@@ -266,7 +266,8 @@ public BisqSetup(DomainInitialisation domainInitialisation,
         this.refundManager = refundManager;
         this.arbitrationManager = arbitrationManager;
 
-        MemPoolSpaceTxBroadcaster.init(socks5ProxyProvider, preferences, localBitcoinNode, config.allowLanForHttpRequests);
+        MemPoolSpaceTxBroadcaster.init(socks5ProxyProvider, preferences, localBitcoinNode,
+                config.allowLanForHttpRequests, config.allowClearnetHttpRequests);
```
[the remark this answers is not in the document]
Comment (wodoro) [reply core/src/main/java/bisq/core/app/BisqSetup.java:270]: User may have legitmate use cases, with good understanding of the risks.

**Review by HenrikJannsen — APPROVED** [review]
utACK
